api/client/client_test.go

```diff
@@ -136,6 +137,38 @@ func TestClient_NewCollection(t *testing.T) {
 	})
 }
 
+func TestIt(t *testing.T) {
+	type Owner struct {
+		ID       core.InstanceID `json:"_id"`
+		ThreadID thread.ID       `json:"threadId"`
```

Comment:
Ok @sanderpick back to the original problem that my original PR of making thread id a string alias would have fixed...

Here we have a model object with a `thread.ID` property. We're going to create a collection and try to save an instance of it.

Reply:
Ah, I see. Okay, let me take a look here in a sec... juggling a few things ATM.